deploy/build.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+
+# Exit on any error
+set -e
+
+# Initialize build type variable as empty to ensure it's set by the user
+BUILD_TYPE=""
+
+# Parse arguments
+while [[ "$#" -gt 0 ]]; do
+    case $1 in
+        -type) BUILD_TYPE="$2"; shift ;;
+        *) echo "Unknown parameter passed: $1"; exit 1 ;;
+    esac
+    shift
+done
+
+# Check if BUILD_TYPE is set, if not, print usage and exit
+if [[ -z "$BUILD_TYPE" ]]; then
+    echo "Usage: $0 -type <BuildType>"
+    echo "Example: $0 -type Debug or $0 -type Release"
+    exit 1
+fi
+
+# Move to the root directory of the project
+cd "$(dirname "$0")/.."
+
+# Load Spack environment
+source ~/Spack/spack/share/spack/setup-env.sh
```

Comment:
Better to check if Spack is installed or not by checking the command `spack` instead of hard coding the path to it. We can direct the users to the prerequisites section of the Wiki if `spack` does not exist.